packages/devtools_app/lib/src/screens/memory/primitives/filter_config.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+const _collectionLibraryUri = 'package:collection';
+const _intlLibraryUri = 'package:intl';
+const _vectorMathLibraryUri = 'package:vector_math';
+
+/// First two libraries are special e.g., dart:* and package:flutter*
+const dartLibraryUriPrefix = 'dart:';
+const flutterLibraryUriPrefix = 'package:flutter';
+
+class FilterConfig {
+  FilterConfig({
+    required this.filterZeroInstances,
+    required this.filterLibraryNoInstances,
+    required this.filterPrivateClasses,
+    required this.libraryFilters,
+  });
+
+  /// Hide any private class, prefixed with an underscore.
+  final ValueNotifier<bool> filterPrivateClasses;
+
+  /// Hide any class that hasn't been constructed (zero instances).
+  final ValueNotifier<bool> filterZeroInstances;
+
+  /// Hide any library with no constructed class instances.
+  final ValueNotifier<bool> filterLibraryNoInstances;
+
+  /// State of filters used by filter dialog (create/modify) and used
+  /// by filtering in grouping.
+  final FilteredLibraries libraryFilters;
+}
+
+/// State of the libraries, wildcard included, filtered (shown or hidden).
+/// groupBy uses this class to determine is the library should be filtered.
```

Comment:
wrap groupBy in brackets if it is a method name `[groupBy]`. You may need to add the class name if the method is not part of this class `[MyClass.groupBy]`